### PR TITLE
[ML] add new `check_frequency` to delayed data check configuration in datafeeds

### PR DESCRIPTION
**Pull request by benwtrent**
Users may want to know about delayed data more quickly. To achieve this, a new `check_frequency` parameter
is provided.

This cannot be less than the datafeed `frequency` as each delayed data check occurs only after the datafeed
has searched and provided data for the anomaly detection job.

### CODE REVIEW

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -458,6 +458,18 @@ that moment in time. See
 This check runs only on real-time {dfeeds}.
 end::delayed-data-check-config[]
 
+tag::delayed-data-check-config-check-frequency[]
+The how often should the {dfeed} search for late data.
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:462]:
```suggestion
Specifies how often the {dfeed} searches for late data.
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 2]
```diff
@@ -458,6 +458,18 @@ that moment in time. See
...
+It must be smaller than the delayed data `check_window`.
+Additionally, if {dfeed} frequency is set, `check_frequency` must
+be larger than `frequency`.
+Delayed data checks cannot occur more often than the configured `frequency`.
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:466]: I think this sentence has the same meaning as the previous sentence
```suggestion

```

File: docs/reference/ml/ml-shared.asciidoc [hunk 3]
```diff
@@ -458,6 +458,18 @@ that moment in time. See
...
+But, setting `check_frequency` higher will increase the time between delayed
+data checks.
+
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:469]: This final paragraph is not being formatted properly in the output. It either needs to be linked to the previous paragraph with a "+" or else just merged into the previous paragraph (I think the latter's my preference)
```suggestion
+
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 4]
```diff
@@ -458,6 +458,18 @@ that moment in time. See
 This check runs only on real-time {dfeeds}.
 end::delayed-data-check-config[]
 
+tag::delayed-data-check-config-check-frequency[]
+The how often should the {dfeed} search for late data.
+It must be smaller than the delayed data `check_window`.
+Additionally, if {dfeed} frequency is set, `check_frequency` must
+be larger than `frequency`.
+Delayed data checks cannot occur more often than the configured `frequency`.
+But, setting `check_frequency` higher will increase the time between delayed
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:467]: If I'm interpreting it correctly, this seems like a "no duh" sentence, so I'd suggest omitting this one too:
> But, setting `check_frequency` higher will increase the time between delayed

**Comment by droberts195** [comment]
I think the only legitimate reason we know of for changing this is when writing tests.  I am just wondering if there's a way to make it possible for tests to run faster without having a whole new datafeed configuration setting that will propagate into docs, UI and client code.

**Comment by pheyos** [comment]
> I think the only legitimate reason we know of for changing this is when writing tests. I am just wondering if there's a way to make it possible for tests to run faster without having a whole new datafeed configuration setting that will propagate into docs, UI and client code.

@droberts195 I can imagine that there are real-world use cases for this configuration as well, e.g. in environments where a prompt reaction to delayed is required, which in turn requires timely alerting. Similar to how we allow to configure the `check_window` I think it's fair to assume that a static delay for the initial report doesn't serve all purposes, although it'll probably work for most customers.

**Comment by benwtrent** [comment]
This work is not strictly necessary. Will proceed in another direction as this setting is not critical for datafeeds.

**Pull request closed without merging**